src/unix_term.rs

```diff
@@ -66,41 +65,76 @@ pub(crate) fn terminal_size(out: &Term) -> Option<(u16, u16)> {
     }
 }
 
-pub(crate) fn read_secure() -> io::Result<String> {
-    let mut f_tty;
-    let fd = unsafe {
-        if libc::isatty(libc::STDIN_FILENO) == 1 {
-            f_tty = None;
-            libc::STDIN_FILENO
+enum Input<T> {
+    Stdin(io::Stdin),
+    File(T),
+}
+
+impl Input<fs::File> {
+    fn new() -> io::Result<Self> {
```

Comment:
As suggested in my earlier comment, I think we should give these `new()` methods proper names, maybe `buffered()` and `unbuffered()`?

Reply:
Sure, done. I made them free functions because otherwise we end up with an unconstrained type parameter and would have to write something like `Input::<fs::File>::unbuffered()` which doesn't look very pleasing.